### ISSUE TITLE
cache-item should resolve initial fetched value in the same fashion as the others

### ISSUE THREAD

**Issue by jmwilkinson** [balihoo/lru-cache-promise#1]
As it stands, the first promise returned by "getAsync" will resolve last. This is because every resolved item is being pushed into an array of resolvables except the last one, which is simply returned. Unfortunately, this results in the odd behavior where the first "getAsync" request is resolved after the other resolvables iterated upon.

The first call should likely behave as the others, going into the resolution array.

This would mean `cache-item.fetch` might look something like the following:

```coffeescript
    fetch: ->
        Promise.try =>
            if @_status is status.fetched or not @fetchFunction
                # Return the current value
                return @value

    # Add a promise to the list of promises awaiting fetch completion
    p = new Promise (resolve, reject) =>
        @resolvers.push resolve
        @rejectors.push reject

    # Call the fetch function
    if @_status is not status.fetching
        Promise.resolve @fetchFunction @key
        .then (value) =>
            @value = value
            @_status = status.fetched

            r value for r in @resolvers
                value

        .catch (err) =>
            @_status = status.fetchFailed
            r err for r in @rejectors

            throw err
    
    @_status = status.fetching

    return p
```

As you can probably tell, I don't really know coffeescript. But hopefully that code is relatively clear. The main thing here is to be consistent with the resolve functions, and not to treat the first one differently.

EDIT: As an addendum to this, after some experimentation, Promise.each would need to be used for the iteration rather than just a for loop. Again, not sure how to do this in coffeescript.

**Comment by balihoo-jflitton** [comment]
Hi there!  Thanks for the feedback.  I've updated the code and released version 1.0.7 to address this issue.

https://github.com/balihoo/lru-cache-promise/commit/baa8b2fc48eb1b7d22d74a4f15fbc50d0746ce69
https://github.com/balihoo/lru-cache-promise/commit/2502e5bbacf147d91b50c5736a9ba8a0995b2387

**Comment by jmwilkinson** [comment]
Thanks! Tested this and it fixed the issue I was seeing. Nice turnaround.
